packages/foundation/src/slots.tsx

```diff
@@ -61,6 +63,8 @@ export function withSlots<P>(
     //        Children had to be spread to avoid breaking KeytipData in Toggle.view:
     //        react-dom.development.js:18931 Uncaught TypeError: children is not a function
     //        Without spread, function child is a child array of one element
+    // TODO: is there a reason this can't be:
+    // return React.createElement.apply(this, arguments);
```

Comment:
this can be what you listed, you should just do that.

Reply:
I did try this before and ran into issues, but I wanted to cut this PR off cleanly before my trip.

Reply:
(Like the other comment, I haven't noticed any negative side effects so these TODOs have been lower priority.)